src/index.ts

```diff
@@ -0,0 +1,59 @@
+import * as core from "@actions/core";
+import { updateCorsConfig } from "./lib/update-cors-cloud";
+import { authenticateCloud } from "./lib/authenticate-cloud";
+
+import { updateClients } from "./lib/update-clients";
+import type { OAuthClientConfig } from "./lib/update-clients";
+
+const AM_URL = core.getInput("AM_URL");
+const username = core.getInput("username");
+const password = core.getInput("password");
+const realm = core.getInput("realm_path");
+const origins = core.getInput("origins"); // this is json as input
```

Comment:
I usually have the habit of appending `Str` to variables that are serialized JSON. That way, the name of the variable hints to its type. So, you get something like this:

```
const originsStr = core.getInput('origins');

const origins = JSON.parse(originsStr);
```

You don't have to use exactly that, but it helps to differentiate the two types at a glance. It also helps with naming variables.